azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/V1ConfigurationParser.java

```diff
@@ -33,15 +34,16 @@ public V1ConfigurationParser(AbstractWebAppMojo mojo) {
     @Override
     public OperatingSystemEnum getOs() throws MojoExecutionException {
         final String linuxRuntime = mojo.getLinuxRuntime();
+        final JavaVersion javaVersion = mojo.getJavaVersion();
         final ContainerSetting containerSetting = mojo.getContainerSettings();
         final boolean isContainerSettingEmpty = containerSetting == null || containerSetting.isEmpty();
 
         // Duplicated runtime are specified
-        if (mojo.getJavaVersion() != null ? linuxRuntime != null || !isContainerSettingEmpty :
+        if (javaVersion != null ? linuxRuntime != null || !isContainerSettingEmpty :
```

Comment:
The logic here is confusing. Can we change to:

```java
int configuredRuntimeNum = 0;
if (javaVersion != null)  configuredRuntimeNum++;
if (linuxRuntime != null)  configuredRuntimeNum++;
if (containerSetting != null && !containerSetting.isEmpty())  configuredRuntimeNum++;
if (configuredRuntimeNum > 1) {
    throw ...
}
```

Reply:
Agree, but still, add a to-do item to refactor the logic, then we are not making the PR bigger and bigger.